twilight-cache-inmemory/src/event/emoji.rs

```diff
@@ -32,7 +32,10 @@ impl InMemoryCache {
     }
 
     pub(crate) fn cache_emoji(&self, guild_id: Id<GuildMarker>, emoji: Emoji) {
-        if let Some(cached_emoji) = self.emojis.get(&emoji.id) {
+        assert!(emoji.id.is_some());
+        let emoji_id = emoji.id.unwrap();
```

Comment:
These two lines do the same, but also I think a better approach is to return early if the emoji doesn't have an `Id`, since it probably shouldn't be cached anyway, with something like
```rust
let Some(emoji_id) = emoji.id else {
    return;
}
```